lib/phoenix_live_view/async.ex

```diff
@@ -3,44 +3,73 @@ defmodule Phoenix.LiveView.Async do
 
   alias Phoenix.LiveView.{AsyncResult, Socket, Channel}
 
-  defp validate_function_env(func, op, env) do
-    warn =
-      # TODO: Remove conditional once we require Elixir v1.14+
-      if Version.match?(System.version(), ">= 1.14.0") do
-        fn msg, env -> IO.warn(msg, env) end
-      else
-        fn msg, _env -> IO.warn(msg) end
-      end
+  defp warn_socket_access(op, warn) do
+    warn.("""
+    you are accessing the LiveView Socket inside a function given to #{op}.
 
-    # prevent false positives, for example
-    # start_async(socket, :foo, function_that_returns_the_anonymous_function(socket))
-    if match?({:&, _, _}, func) or match?({:fn, _, _}, func) do
-      Macro.prewalk(func, fn
-        {:socket, meta, nil} ->
-          warn.(
-            """
-            you are accessing the LiveView Socket inside a function given to #{op}.
+    This is an expensive operation because the whole socket is copied to the new process.
+
+    Instead of:
 
-            This is an expensive operation because the whole socket is copied to the new process.
+        #{op}(socket, :key, fn ->
+          do_something(socket.assigns.my_assign)
+        end)
 
-            Instead of:
+    You should do:
 
-                #{op}(socket, :key, fn ->
-                  do_something(socket.assigns.my_assign)
-                end)
+        my_assign = socket.assigns.my_assign
 
-            You should do:
+        #{op}(socket, :key, fn ->
+          do_something(my_assign)
+        end)
+
+    For more information, see https://hexdocs.pm/elixir/1.16.1/process-anti-patterns.html#sending-unnecessary-data.
+    """)
+  end
 
-                my_assign = socket.assigns.my_assign
+  # this is not private to prevent the unused function warning as we only
+  # call this function when enable_expensive_runtime_checks is set
```

Comment:
```suggestion
  # call this function when enable_expensive_runtime_checks is set
  @doc false
```

Reply:
does the `@doc false` do anything when `@moduledoc false` is already set? Should we add `@doc false` to the other functions as well?

Reply:
No need. Ignore this. :)